### PR TITLE
Fixed working of shaders on AMD cards in OGL3 mode.

### PR DESCRIPTION
**Pull request by JavaSaBr**
Fixed shaders on AMD cards in the OGL3 mode.
For example:
```
WARNING 17:45:24:823 Editor: RendererException : compile error in: ShaderSource[name=Common/MatDefs/Misc/Particle.frag, defines, type=Fragment, language=GLSL120]
Fragment shader failed to compile with the following errors:
ERROR: error(#271) Explicit version number 120 not supported by GL3 forward compatible context
ERROR: error(#273) 1 compilation errors.  No code generated

WARNING 17:47:44:276 Editor: RendererException : compile error in: ShaderSource[name=Common/MatDefs/Light/PBRLighting.frag, defines, type=Fragment, language=GLSL110]
Fragment shader failed to compile with the following errors:
ERROR: error(#272) Implicit version number 110 not supported by GL3 forward compatible context
ERROR: error(#273) 1 compilation errors.  No code generated

WARNING 17:56:34:912 Editor: RendererException : compile error in: ShaderSource[name=Common/MatDefs/Blur/HGaussianBlur.frag, defines, type=Fragment, language=GLSL100]
Fragment shader failed to compile with the following errors:
ERROR: error(#272) Implicit version number 110 not supported by GL3 forward compatible context
ERROR: error(#273) 1 compilation errors.  No code generated

	
WARNING 17:57:42:794 Editor: RendererException : compile error in: ShaderSource[name=Common/MatDefs/Blur/VGaussianBlur.frag, defines, type=Fragment, language=GLSL100]
Fragment shader failed to compile with the following errors:
ERROR: error(#272) Implicit version number 110 not supported by GL3 forward compatible context
ERROR: error(#273) 1 compilation errors.  No code generated

WARNING 17:58:25:357 Editor: RendererException : compile error in: ShaderSource[name=Common/MatDefs/Hdr/LogLum.frag, defines, type=Fragment, language=GLSL100]
Fragment shader failed to compile with the following errors:
ERROR: error(#272) Implicit version number 110 not supported by GL3 forward compatible context
ERROR: error(#273) 1 compilation errors.  No code generated
```

### CODE REVIEW

**Comment by JavaSaBr** [comment]
I have updated this PR.

**Comment by JavaSaBr** [comment]
I have added sorting by GLSL version to select the best tech.

File: jme3-core/src/main/resources/Common/MatDefs/Light/PBRLighting.frag [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#extension GL_NV_shadow_samplers_cube : enable
+
```
Comment (Nehon) [inline jme3-core/src/main/resources/Common/MatDefs/Light/PBRLighting.frag]: Why is this needed?
1. this is a NVidia only extension I wonder why it's in a fix for AMD cards
2. It's an extention that allows you to fetch into cube shadow maps. Things we don't do at all, and if we did it wouldn't be in this shader.
